src/evo/creditpool.cpp

```diff
@@ -114,17 +114,20 @@ void CCreditPoolManager::AddToCache(const uint256& block_hash, int height, const
 
 static std::optional<CBlock> GetBlockForCreditPool(const CBlockIndex* const block_index, const Consensus::Params& consensusParams)
 {
+    // There's no CbTx before DIP0003 activation
+    if (!DeploymentActiveAt(*block_index, Params().GetConsensus(), Consensus::DEPLOYMENT_DIP0003)) {
```

Comment:
block_index can be null here?

Reply:
it seems unrelated because it was used in `ReadBlockFromDisk` before without extra checks:
```
bool ReadBlockFromDisk(CBlock& block, const CBlockIndex* pindex, const Consensus::Params& consensusParams)
{   
    const FlatFilePos block_pos{WITH_LOCK(cs_main, return pindex->GetBlockPos())};
```

Anyway, I added extra annotations just in case, see a new commit